internal/engine/upper.go

```diff
@@ -727,6 +727,9 @@ func handleAnalyticsNudgeSurfacedAction(ctx context.Context, state *store.Engine
 }
 
 func handleTiltCloudUserLookedUpAction(state *store.EngineState, action store.TiltCloudUserLookedUpAction) {
+	if state.TiltCloudUsername != "" {
+		state.TiltCloudUsernameNeedsRefresh = false
```

Comment:
I think I've solved this now by adding `action.IsPostRegistrationLookup`.

This also caught a related race condition:
1. CloudUsernameManager starts doing a normal username lookup
2. We trigger a long get lookup. CloudUsernameManager does nothing because there's already a request in process.
3. The lookup from (1) completes, setting TiltCloudUsernameNeedsRefresh to false
4. We never do the lookup from (2).